data/tutorials/language/0lg_00_basic_datatypes.md

```diff
@@ -0,0 +1,808 @@
+```
+
+Defining a function is the same as naming a value, as illustrated in the first expression:
+
+```ocaml
+# let g x = x * x;;
+val g : int -> int = <fun>
+
+# g 9;;
+- : int = 81
+```
+
+Executable OCaml code consists primarily of functions, so it's beneficial to make them as concise and clear as possible. The function `g` is defined here using a shorter, more common, and maybe more intuitive syntax.
+
+In OCaml, functions may terminate without returning the expected type value by throwing an exception (`exn`), which does not appear in its type. There is no way to know if a function may raise an exception without inspecting its code.
+```ocaml
+# raise;;
+- : exn -> 'a' = <fun>
+```
+
+Exceptions are discussed in the [Error Handling](/docs/error-handling) guide.
+
+Functions may have several parameters.
+```ocaml
+# fun a b -> a ^ " " ^ b;;
+- : string -> string -> string = <fun>
+
+# let mean a b = (a + b) / 2;;
+val mean : int -> int -> int = <fun>
+```
+
+Like the product type symbol `*`, the function type symbol `->` is not associative. The following two types are not the same:
+- `(int -> int) -> int` : This function takes a function of type `int -> int` as a parameter and returns an `int` as the result.
+- `int -> (int -> int)` : This function takes an `int` asa  parameter and returns a function of type `int -> int` as the result.
+
+### Unit
+
+Uniquely, the type `unit` has only one value. It is written `()` and pronounced “unit.”
+
+The `unit` type has several uses. Mainly, it serves as a token when a function does not need to be passed data or doesn't have any data to return once it has completed its computation. This happens when functions have side effects such as OS-level I/O. Functions need to be applied to something for their computation to be triggered, and they also must return something. When nothing meaningful can be passed or returned, `()` should be used.
+```ocaml
+# read_line;;
+- : unit -> string = <fun>
+
+# print_endline;;
+- : string -> unit = <fun>
+```
+
+The function `read_line` reads an end-of-line terminated sequence of characters from standard input and returns it as a string. Reading input begins when `()` is passed.
+
+The function `print_endline` prints the string followed by a line ending on standard output. Return of the unit value means the output request has been queued by the operating system.
+
+## User-Defined Types
+
+User defined types are always introduced using the `type … = …` construction. The keyword `type` must written in lowercase. The first ellipsis stands for type name and must not begin by a capital. The second ellipsis stands for the type definition. Three cases are possible:
+1. Variant
+1. Record
+1. Aliases
+
+These three kinds of type definitions are covered in next three sections.
+
+### Variants
+
+Variants are also called [_tagged unions_](https://en.wikipedia.org/wiki/Tagged_union). They relate to the concept of [_disjoint union_](https://en.wikipedia.org/wiki/Disjoint_union).
```

Comment:
Do you want an example here like you have for Enumerated, etc.?

Reply:
This is an introduction for the rest of this section. Examples will come later.